crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,68 @@
+use envconfig::Envconfig;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    pub otel_sdk_disabled: bool,
+
+    /// Determines the verbosity of the OpenTelemetry tracing layer.
+    ///
+    /// This is a Spin specific value. This is what allows us to have different levels of verbosity
+    /// in the fmt layer and the otel layer. The fmt layer respects `RUST_LOG` and the otel layer
+    /// respects `SPIN_OTEL_TRACING_LEVEL`.
+    pub otel_tracing_level: String,
+
+    /// Spin version.
+    pub version: String,
```

Comment:
Is this name forced on us by OpenTelemetry, or could we rename it to something more specific such as `spin_version`?

Reply:
The underlying thing this maps to is `system.version` which is forced on us by OTEL. But, this we could certainly give a nicer name like `spin_version`. Will do.